include/libsupl.rst

```diff
@@ -0,0 +1,153 @@
+
+A standalone GPS uses the orbital data (like `Ephermeridus`_ and the `Almanac`_) of the satellite to calculate the current position of a device.
+Typically, the data download from the satellite takes a long time and the first position is calculated in 30-40 seconds.
+Any loss of signal requires the whole process to be restarted.
+
+An A-GPS server is a cached server that stores the orbital data.
+This information can be downloaded at a fast rate since an A-GPS capable device supports mobile radio-network bearers with high data rate such as GSM or LTE.
+
+The SUPL version supported by this library is 2.0.4.
+
+LibSUPL payload
+***************
+
+`LPP (LTE Positioning Protocol)`_ is the supported SUPL payload format.
+
+LPP is a point-to-point protocol used between a location server (for example, SLP) and a target device (for example, SET) to position the device using position-based information obtained from reference sources.
+A SET or SUPL Enabled Terminal is a logical part of a device that supports communication with a SUPL network.
+
+LibSUPL session and transaction type
+************************************
+
+A typical libSUPL session is SET-based, i.e. the position is calculated based on SET.
+See `SET-initiated SUPL collaboration`_ for more information.
+
+Each LPP session consists of one or more LPP transactions.
+Each LPP transaction corresponds to a single operation such as capability exchange, assistance data transfer, or location information transfer.
+LibSUPL supports only assistance data transfer as the transaction type.
+
+The following image shows the steps in the libSUPL connection session.
+
+.. figure:: /images/libSUPL_connection_session.png
+      :alt: LibSUPL connection session
+
+LibSUPL architecture
+********************
+
+The libSUPL architecture consists of modules (both internal and external) and API callback functions.
+These are described in the subsequent sections.
+
+Modules
+=======
+
+There are several modules inbuilt into the LibSUPL architecture as shown in the figure below.
+Additionally, the library makes use of the external module :ref:`ossnokalva_asn1_lib`.
+
+.. figure:: /images/libsupl_module_architecture.png
+      :alt: LibSUPL module architecture
+
+Resource initialization and ownership
++++++++++++++++++++++++++++++++++++++
+
+The buffers used in the SUPL communication session, namely the primary and secondary buffers, are owned by the user of the libSUPL library.
+The buffers are characterized by the sizes defined by LIBSUPL_PRI_BUF_SIZE and LIBSUPL_SEC_BUF_SIZE.
+These parameters are defined in the ``supl_session.h`` header file.
+The buffers are passed to libSUPL through the supl_client_init() function.
+
+Communication with the SUPL server must be handled by the application through the read and write callback functions.
+
+SUPL Flow Control
++++++++++++++++++
+
+LibSUPL is initialized by calling the supl_init() function before the first SUPL session.
```

Comment:
:cpp:func:`supl_init` *should* work (same for supl_session).

Reply:
Made this change for all functions specified.